WebClient/Scripts/temperatureControlView.js

```diff
@@ -0,0 +1,503 @@
+        //this._draggable = new L.Draggable(this._container);
+        //this._draggable.ref = this;
+
+        //this._draggable.enable();
+
+        this._temperatureDiv = L.DomUtil.create('div', "temperatureDiv");
+        this._container.appendChild(this._temperatureDiv);
+
+        //disabling propagation
+        L.DomEvent.disableClickPropagation(this._container);
+        L.DomEvent.disableScrollPropagation(this._container);
+
+        // use context menu event to return to "live" state
+        this._container.addEventListener('contextmenu', this._handleContextMenu);
+
+        //--------------------Variable declarations------------------------
+        //svg image vars
+        var width = 90;
+        var height = 160;
+        this.borderWidth = 1;
+
+        //general x-coordinate var
+        this.xPosForBaseComponents = (width / 2.5);
+
+        //capillary tube vars
+        this.tubeWidth = 12;
+        this.tubeRectY = (height / 8);
+        this.tubeHeight = ((3.1 * height) / 4) - this.tubeRectY;
+        this.tubeRectX = this.xPosForBaseComponents - (this.tubeWidth / 2);
+
+        this.tubeBulbPixelOverlap = 3; //the spherical bulb overlaps the capillary tube by defined pixels
+
+        //mercury vals
+        this.mercuryColor_red = "#E50000";
+        this.mercuryColor_blue = "#30669F";
+        this.centerColor = "#CCD1D1";
+
+        //min and max temperature in K
+        var minTemperature = new UnitConverter.ConvNum('Temperature', 253.15);
+        var maxTemperature = new UnitConverter.ConvNum('Temperature', 323.15);
+
+        //kelvin min and max values
+        this.minKelvin = minTemperature.GetDisplayValue();
+        this.maxKelvin = maxTemperature.GetDisplayValue();
+
+        //celsius min and max values
+        minTemperature.SetUnit('°C');
+        maxTemperature.SetUnit('°C');
+        this.minCelsius = parseFloat(minTemperature.GetDisplayValue().toFixed(2));
+        this.maxCelsius = parseFloat(maxTemperature.GetDisplayValue().toFixed(2));
+
+        //fahrenheit min and max values
+        minTemperature.SetUnit('°F');
+        maxTemperature.SetUnit('°F');
+        this.minFahrenheit = parseFloat(minTemperature.GetDisplayValue().toFixed(2));
+        this.maxFahrenheit = parseFloat(maxTemperature.GetDisplayValue().toFixed(2));
+
+        //Universal Scale
+        this.universalScale = d3.scale.linear()
+            .domain([this.minKelvin, this.maxKelvin])
+            .range([this.tubeHeight - this.tubeBulbPixelOverlap, 0]);
+
+        /*
+        //D3 V4 code for the scale
+        this.universalScale = d3.scaleLinear()
+            .domain([this.minKelvin, this.maxKelvin])
+            .range([this.tubeHeight - this.tubeBulbPixelOverlap, 0]);
+        */
+
+        //selected unit variable - default set to C
+        this.selectedUnit = TemperatureUnit.C;
+
+        //setting up the temperature model and passing in the default values
+        this.temperatureModel = new TemperatureModel({
```

Comment:
In the future you'll have to connect this component with the server therefore you'll have a separate object to handle that. In that case you'll need to share the model with that object. Therefore it is better to create a model from a different external place and to pass it to this view. In such case View will be only responsible for representing a model and will not decide on how to create a model and how to link it with other parts of code.

FYI, take a look Dependency Injection pattern